src/Sylius/Behat/Page/Shop/Product/IndexPage.php

```diff
@@ -58,35 +59,41 @@ public function sort(string $orderNumber): void
 
     public function clearFilter(): void
     {
-        $this->getDocument()->clickLink('Clear');
+        $this->getElement('clear')->click();
     }
 
     public function isProductOnList(string $productName): bool
     {
-        return null !== $this->getDocument()->find('css', sprintf('.sylius-product-name:contains("%s")', $productName));
+        try{
```

Comment:
why did you change it to try/catch block?

Reply:
when I replace getDocument() to getElement() and element will not exist, getElement throw exception, we can use hasElement if all element is defined like hasElement('product') but if we try hasElement('product', ['%productName% => $productName']) we get error